SPEC/PIN.md

```diff
@@ -53,28 +50,24 @@ A great source of [examples][] can be found in the tests for this API.
 Where:
 
 - `cid` - a [CID][cid] instance or CID as a string or an array of CIDs.
-- `options` is an object that can contain the following keys:
-  - 'type' - Return also the type of pin (direct, indirect or recursive)
+- `options` - is an object that can contain the following keys:
+  - `type` - filter by this type of pin ("recursive", "direct" or "indirect")
 
 **Returns**
 
 | Type | Description |
 | -------- | -------- |
-| `Promise<Array>` | An array of current pinned objects |
-
-an array of objects with keys `hash` and `type` is returned.
+| `AsyncIterable<{ hash: string, type: string }>` | An async iterable that yields currently pinned objects with `hash` and `type` properties. `hash` is a string CID of the pinned node, `type` is the pin type ("recursive", "direct" or "indirect") |
```

Comment:
I think `hash` should be replaced with `cid`, to match https://github.com/ipfs/js-ipfs-http-client/pull/1183:

> BREAKING CHANGE: pin.ls results now contain a cid property (a CID instance) instead of a string hash property.